### PR TITLE
Accumulate: Added exercise definition files for accumulate

### PR DESCRIPTION
**Pull request by KalebHawkins**
Really not sure if I got all this right the first time around or not. This is the first exercise I have submitted. I do feel like there could be plenty more test cases may add some of those later. Please let me know if there is anything I can improve on.

### CODE REVIEW

**Comment by kotp** [comment]
It looks like there are some "non-exercise" related commits included as well.  I noticed that configlet binary is added in this PR as well.

Probably want one PR to set up the track tooling, and one for the exercise.

Looks like things are coming together though!  Keep up the good work.

File: bin/fetch-configlet [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/usr/bin/env bash
```
Comment (ErikSchierboom) [inline bin/fetch-configlet:1]: A good rule for PRs is to have them be as specific as possible. This PR is about adding an exercise, which means that ideally the changes to this file should not be part of this PR. Could you revert these changes to this file?

File: bin/run-tests.ps1 [hunk 2]
```diff
@@ -1,5 +1,5 @@
 $tmpRoot = [System.IO.Path]::GetTempPath()
-$exercises = Get-ChildItem -Path exercises/practice -Directory
+$exercises = Get-ChildItem -Path ../exercises/practice -Directory
```
Comment (ErikSchierboom) [inline bin/run-tests.ps1:2]: Same comment as on the above file.

File: config.json [hunk 3]
```diff
@@ -151,10 +151,30 @@
...
   "key_features": [],
-  "tags": []
+  "tags": [
+    "paradigm/procedural",
+    "typing/dynamic",
+    "execution_mode/interpreted",
+    "platform/windows",
+    "platform/mac",
+    "platform/linux"
+  ]
```
Comment (ErikSchierboom) [inline config.json:179]: Same comment regarding PRs to be as focused as possible.

File: exercises/practice/accumulate/.meta/config.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "blurb": "Implement the `accumulate` operation, which, given a collection and an operation to perform on each element of the collection, returns a new collection containing the result of applying that operation to each element of the input collection.",
+  "authors": [
+    "Kryo"
+  ],
+  "contributors": [
+    "Kryo"
+  ],
```
Comment (ErikSchierboom) [inline exercises/practice/accumulate/.meta/config.json:8]: Someone is either an author or a contributor, but not both. You're definitely the author :)

```suggestion
  "contributors": [],
```

File: exercises/practice/hello-world/GETTING_STARTED.md [hunk 5]
```diff
@@ -2,8 +2,13 @@
...
 
-The following steps assume that you are in the same directory as the test
-suite.
+For information installing Powershell click [here](https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell?view=powershell-7.1).
+
+Once Powershell is installed you will need to install the Pester module by running the following command in Powershell:
+
+    PS> Install-Module Pester
+
+The following steps assume that you are in the same directory as the test suite.
```
Comment (ErikSchierboom) [inline exercises/practice/hello-world/GETTING_STARTED.md:11]: Same comment regarding PRs to be focused.

File: exercises/shared/.docs/cli.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# CLI (Command Line Interface)
```
Comment (ErikSchierboom) [inline exercises/shared/.docs/cli.md:1]: Same comment regarding PRs to be focused.

File: exercises/shared/.docs/debug.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Debug
```
Comment (ErikSchierboom) [inline exercises/shared/.docs/debug.md:1]: Same comment regarding PRs to be focused.

File: exercises/shared/.docs/help.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# Help
```
Comment (ErikSchierboom) [inline exercises/shared/.docs/help.md:1]: Same comment regarding PRs to be focused.

File: exercises/shared/.docs/tests.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# Tests
```
Comment (ErikSchierboom) [inline exercises/shared/.docs/tests.md:1]: Same comment regarding PRs to be focused.

File: exercises/practice/accumulate/description.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
+# Description
```
Comment (ErikSchierboom) [inline exercises/practice/accumulate/description.md:1]: A practice exercise does not have a `description.md` file. See [the docs](https://exercism.org/docs/building/tracks/practice-exercises). Could you remove this file?

File: exercises/practice/accumulate/Get-Accumulation.tests.ps1 [hunk 11]
```diff
@@ -0,0 +1,23 @@
+BeforeAll {
+    . "./Get-Accumulation.ps1"
+}
+
+Describe 'Test Get-Accumulation.ps1' {
```
Comment (ErikSchierboom) [inline exercises/practice/accumulate/Get-Accumulation.tests.ps1:5]: These tests look to be a subset of the tests in the canonical data of this exercise: https://github.com/exercism/problem-specifications/blob/main/exercises/accumulate/canonical-data.json Was there a specific reason for that?

File: exercises/practice/accumulate/.meta/example.ps1 [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        [Int64[]]$List,
+        [scriptblock]$Func
+    )
+
+    $Func.Invoke($List)
+}
+
+Get-Accumulation -List 1,2,3,4,5 -Func { Param([Int64[]]$numbers) $numbers | ForEach-Object { $_ * $_ }}
+
+Get-Accumulation -List 1,2,3,4,5 -Func { Param([Int64[]]$numbers) $numbers | ForEach-Object { $_ + $_ }}
```
Comment (ErikSchierboom) [inline exercises/practice/accumulate/.meta/example.ps1:41]: I don't think these calls should be in the example file, as they're the tests and belong to the tests file.

```suggestion
```

**Review by ErikSchierboom — COMMENTED** [review]
Hi! Thanks for working on this. I've left some comments, most of which involve this PR updating files unrelated to the accumulate exercise. Let me know if you'd like help fixing things based on my comments.

A practice exercise should have a `.meta/tests.toml` file. You can generate this via configlet:

```
./bin/fetch-configlet
./bin/configlet sync --exercise accumulate --update --mode include
```

**Comment by KalebHawkins** [comment]
I'm not sure why all the extra commits are there. I guess when I branched it didn't branch from main and it branched from my dev branch. Still getting used to the way git works other than basic operations sorry about that. I will probably close this PR and rebase honestly.

**Comment by kotp** [comment]
Rebasing is fine, and force pushing in your branches is also not only OK, but expected.

Get with me if you would like to explore some git stuff with me.
